add_varsel_cmap.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python
+#
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Modify an Emoji font to populate cmap 14 variation selector subtable.
+
+This uses the unicode variation selector data, finds all characters in the
+standard cmap that match, and generates a cmap format 14 subtable indicating
+that they have the provided presentation (emoji by default).  No fonts
+are processed if any already has a format 14 subtable.  Modified fonts
+are written to the provided dir and optionally have a provided suffix
+appended (before the extension).  An output file name can be provided
+in which case only one input file is allowed."""
+
+import argparse
+import os
+from os import path
+import re
+import sys
+
+from fontTools import ttLib
+from fontTools.ttLib.tables import _c_m_a_p as CMap
+
+from nototools import font_data
+
+UCD = path.join(path.dirname(__file__), 'third_party', 'ucd')
+
+def get_emoji_data():
+  """Parse emoji-data.txt and return a tuple of two sets of characters:
+  - those with a default emoji presentation
+  - those with a default text presentation"""
+
+  presentation_default_text = set()
+  presentation_default_emoji = set()
+  emoji_data = path.join(UCD, 'emoji-data.txt')
+  line_re = re.compile(r'([0-9A-F]{4,6})\s*;\s*(emoji|text)\s*;.*')
+  with open(emoji_data, 'r') as f:
+    for line in f.readlines():
+      m = line_re.match(line)
+      if m:
+        cp = int(m.group(1), 16)
+        if m.group(2) == 'emoji':
+          presentation_default_emoji.add(cp)
+        else:
+          presentation_default_text.add(cp)
+  return presentation_default_emoji, presentation_default_text
+
+
+def get_unicode_emoji_variants():
+  """Parse StandardizedVariants.txt and return a set of characters
+  that have a defined emoji variant presentation.  All such characters
+  also have a text variant presentation (as of v8.0.0), so a single
```

Comment:
The characters also having a text variant presentation is expected to continue forever. Feel free to mention that and/or remove the 8.0.0 reference.